src/tools/compiletest/src/json.rs

```diff
@@ -9,15 +9,15 @@
 // except according to those terms.
 
 use errors::{Error, ErrorKind};
-use rustc_serialize::json;
+use serde_json;
 use std::str::FromStr;
 use std::path::Path;
 use runtest::ProcRes;
 
 // These structs are a subset of the ones found in
 // `syntax::json`.
 
-#[derive(RustcEncodable, RustcDecodable)]
+#[derive(Serialize, Deserialize)]
```

Comment:
Does deriving `Serialize` really need? I did the same thing on [`compiletest-rs`](https://github.com/laumann/compiletest-rs/pull/104) crate several days ago and found that it's not needed for that crate. Not sure about this in-tree version though.

Reply:
Excellent point!